sdk/resources/arm-resources/src/resourceManagementClientContext.ts

```diff
@@ -1,74 +1,62 @@
- * regenerated.
- */
+import * as coreClient from "@azure/core-client";
+import * as coreAuth from "@azure/core-auth";
+import { ResourceManagementClientOptionalParams } from "./models";
 
-import * as Models from "./models";
-import * as msRest from "@azure/ms-rest-js";
-import { TokenCredential } from "@azure/core-auth";
-import * as msRestAzure from "@azure/ms-rest-azure-js";
-
-const packageName = "@azure/arm-resources";
-const packageVersion = "4.2.0";
-
-export class ResourceManagementClientContext extends msRestAzure.AzureServiceClient {
-  credentials: msRest.ServiceClientCredentials | TokenCredential;
+export class ResourceManagementClientContext extends coreClient.ServiceClient {
+  $host: string;
+  apiVersion: string;
   subscriptionId: string;
-  apiVersion?: string;
 
   /**
-   * Initializes a new instance of the ResourceManagementClient class.
-   * @param credentials Credentials needed for the client to connect to Azure. Credentials
-   * implementing the TokenCredential interface from the @azure/identity package are recommended. For
-   * more information about these credentials, see
-   * {@link https://www.npmjs.com/package/@azure/identity}. Credentials implementing the
-   * ServiceClientCredentials interface from the older packages @azure/ms-rest-nodeauth and
-   * @azure/ms-rest-browserauth are also supported.
-   * @param subscriptionId The ID of the target subscription.
-   * @param [options] The parameter options
+   * Initializes a new instance of the ResourceManagementClientContext class.
+   * @param credentials Subscription credentials which uniquely identify client subscription.
+   * @param subscriptionId The Microsoft Azure subscription ID.
+   * @param options The parameter options
    */
   constructor(
-    credentials: msRest.ServiceClientCredentials | TokenCredential,
+    credentials: coreAuth.TokenCredential,
     subscriptionId: string,
-    options?: Models.ResourceManagementClientOptions
+    options?: ResourceManagementClientOptionalParams
   ) {
-    if (credentials == undefined) {
-      throw new Error("'credentials' cannot be null.");
+    if (credentials === undefined) {
+      throw new Error("'credentials' cannot be null");
     }
-    if (subscriptionId == undefined) {
-      throw new Error("'subscriptionId' cannot be null.");
+    if (subscriptionId === undefined) {
+      throw new Error("'subscriptionId' cannot be null");
     }
 
+    // Initializing default values for options
     if (!options) {
       options = {};
     }
-    if (!options.userAgent) {
-      const defaultUserAgent = msRestAzure.getDefaultUserAgentValue();
-      options.userAgent = `${packageName}/${packageVersion} ${defaultUserAgent}`;
+    const defaults: ResourceManagementClientOptionalParams = {
+      requestContentType: "application/json; charset=utf-8",
+      credential: credentials
+    };
+
+    const packageDetails = `azsdk-js-arm-resources/1.0.0-beta.1`;
```

Comment:
@sarangan12  package version is wrong here.

Reply:
The package version is changed by my tools, instead of codegen. I will fix it.